packages/@aws-cdk/custom-resource-handlers/lib/aws-events-targets/aws-api-handler/index.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable no-console */
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { getV3Client, getV3Command, loadV3ClientPackage } from '@aws-cdk/sdk-v2-to-v3-adapter';
+
+interface AwsApiInput {
+  readonly service: string;
+  readonly action: string;
+  readonly parameters?: any;
+  readonly apiVersion?: string;
+  readonly catchErrorPattern?: string;
+}
+
+export async function handler(event: AwsApiInput) {
+  console.log('Event: %j', { ...event, ResponseURL: '...' });
+
+  const awsSdk = loadV3ClientPackage(event.service);
+  console.log(`AWS SDK V3: ${awsSdk.packageName}@${awsSdk.packageVersion}`);
+
+  const client = getV3Client(awsSdk, { apiVersion: event.apiVersion });
+  const Command = getV3Command(awsSdk, event.action);
+  const commandInput = event.parameters ?? {};
+
+  try {
+    const response = await client.send(new Command(commandInput));
+    delete response.$metadata;
+    console.log('Response: %j', response);
```

Comment:
I get that this handler is for triggering effects and doesn't return a value, but I still feel like there might be a sharp edge here wrt Uint8Array types. Someone could use this to trigger Kinesis today passing a string payload and that would break right? We just pass parameters as aws-events passes them I suppose, and I'm not even sure how we could place a marker or something for special case deserialization.

Reply:
Let me double check this,. But I'm quite sure the default logging algorithm does work for every object without fail. Unlike JSON.stringify() which often throws for unknown data.

Reply:
Checked this. It logs the payload as bytes, `{"0":123,"1":34,"2":101,"3":114,"4":114,"5":111,"6":114,"7":84,"8":121, ...` and doesn't fail. It's not great, but I think it's okay.

As for input: The function is called from EvenBridge (and the API directly) which is JSON by definition. Any other invocations would be outside the expected usage pattern and transformation into a string would be the users responsibility.

Reply:
I think Mitch' concern is the opposite?

If the SDKv2 API used to accept a `string` and pass it on, but SDKv3 API now accepts a `Uint8Array` for that same field (but not a `string`), existing use cases are now broken.

Reply:
Ah I see. Yes you are right about this. 

> It would seem like the only way to do this properly is to know exactly what fields are marked as "blob"s in the API

Yes and that seems unreasonable for us to do.

Reply:
That seems unreasonable, yes. But right now this is breaking certain use cases.